src/main/resources/schema/elytron-client-1_1.xsd

```diff
@@ -252,8 +252,11 @@
     </xsd:complexType>
 
     <xsd:complexType name="resource-owner-credentials-type">
+        <xsd:choice minOccurs="0" maxOccurs="1">
+            <xsd:element name="credential-store-reference" type="credential-store-reference-type" />
+        </xsd:choice>
         <xsd:attribute name="name" type="xsd:string" use="required" />
-        <xsd:attribute name="password" type="xsd:string" use="required" />
+        <xsd:attribute name="password" type="xsd:string" use="optional" />
```

Comment:
When credential-store-reference is available, there is no need to provide password attribute too - clear-text credential reference can be used:

    credential-reference={clear-text="randall"}